### ISSUE TITLE
bug(Datepicker): In multi-year view, date picker doesnt announce selected year range

### ISSUE THREAD

**Issue by Kmaczek** [angular/components#28360]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

14.2.7

### Description

Steps to reproduce:
1. Open multi year view.
![image](https://github.com/angular/components/assets/1743813/1ad1c08f-8d7c-4ebb-b7d3-51a0e0b7efad)
3. Tab to previous (or next) arrow and press enter.
![image](https://github.com/angular/components/assets/1743813/2b0d5fcd-6cf2-44b2-bb0f-8e8d8fc64e4b)
4. Observe that, displayed year range is updated, but it's not announced
5. Tab back to the year range (incorrectly announced for firefox)
![image](https://github.com/angular/components/assets/1743813/18e19d9e-8950-4301-8491-fc957b5b0bb3)

![image](https://github.com/angular/components/assets/1743813/1c7e28c5-7c4c-4adb-a9e9-f12fb90d68f8)

### Reproduction

Can be reproduced on documentation page: https://material.angular.io/components/datepicker/overview#accessibility with basic datepicker example (first one)
I checked it for versions 17.0.4 and 16.2.12, 15.2.9, 14.2.7 and 12.2.13

I used NVDA, Jaws, Firefox, Chrome

### Expected Behavior

- **Related to Repro step no. 3**: Announce change of year range. 
![image](https://github.com/angular/components/assets/1743813/f2ecd099-9e9d-42e2-87ec-cc55c43034ee)

- **Related to Repro step no. 4**: Announce '1992 to 2015'

### Actual Behavior

- **Related to Repro step no. 3**: Visual changes should be announced to the user. This doesn't happen since version 15.
Happens for both JAWS/NVDA and Chrome/Firefox

- **Related to Repro step no. 4**:  its announcing previous range '2016 to 2039', instead announcing '1992 to 2015' according to displayed label. This happens for Firefox only.

### Environment

- Angular: 16.1.3 (any after 14)
- CDK/Material: 16.1.3 (any after 14)
- Browser(s): Chrome, Firefox
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by Kmaczek** [comment]
Hi @zarend, it doesn't happen for NVDA indeed, but I was able to reproduce below with JAWS + Chrome/Firefox.

> Related to Repro step no. 3: Visual changes should be announced to the user. This doesn't happen since version 15.
> Happens for ~~both~~ JAWS ~~/NVDA~~ and Chrome/Firefox

    material.angular.io
    Windows 11 Pro Version 22H2
    Chrome Version 120.0.6099.200 (Official Build) (64-bit)
    Firefox 121.0.1 (64-bit)
    JAWS Version 2023.2212.23 ILM
